File: Aditya Pratap SIngh/PrefixSumAssessment/Problem1.py
def middleindex(nums):
    n = len(nums)
    totalsum = sum(nums)
    leftsum = 0
    
    for i in range(n):
        if leftsum == (totalsum - leftsum - nums[i]):
            return i
        leftsum += nums[i]
    
    return -1

#nums = [2, 3, -1, 8, 4]
nums = [1,-1, 4]
res = middleindex(nums)
print(res)
#Time Complexity = O(n) 
#Space Complexity = O(1)

File: Aman Shukla/Problem1.java
// Time Complexity = O(n)
// Space Complexity = O(1)
import java.util.Scanner;

public class seesawBalance {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        System.out.print("Enter the number of elements in the array: ");
        int n = sc.nextInt();
        int[] nums = new int[n];
        System.out.println("Enter the elements of the array: ");
        for (int i = 0; i < n; i++) {
            nums[i] = sc.nextInt();
        }

        int totalSum = 0;
        for (int num : nums) {
            totalSum += num;
        }

        int leftSum = 0;
        boolean found = false;
        for (int i = 0; i < nums.length; i++) {
            if (leftSum == totalSum - leftSum - nums[i]) {
                System.out.println(i);
                found = true;
                break;
            }
            leftSum += nums[i];
        }
        if (!found) {
            System.out.println("-1");
        }
    }
